File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    
    public String solution(int[] numbers) {
        ArrayList<String> strList = new ArrayList<>();
        for (int num : numbers) strList.add(String.valueOf(num));
        
        Collections.sort(strList, (s1, s2) -> {
            int a = Integer.parseInt(s1 + s2);
            int b = Integer.parseInt(s2 + s1);
            return Integer.compare(b, a);
        });
        
        StringBuilder sb = new StringBuilder();
        for (String str : strList) sb.append(str);
        return sb.charAt(0) == '0' ? "0" : sb.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    answer = ''
    numbers = list(map(str, numbers))
    numbers.sort(key=cmp_to_key(lambda x, y: int(x+y)-int(y+x)), reverse=True)
    return str(int(''.join(numbers)))